### PR TITLE
Refactor if/else ladder to make it more readable

### LINKED ISSUE #44: Running "pasta analyse succ" in mbox mode doesnt show appropriate error message
Analysing successive patches should only be possible in patch stack mode.
Currently, if I run `patch analyse succ` with Linux as my selected configuration this is the output:
```
2020-03-18 00:28:48,876 PaStA           INFO     Cmdline: ./pasta analyse succ
2020-03-18 00:28:48,876 pypasta.Config  INFO     Active configuration: linux
2020-03-18 00:28:48,931 Repository.Mbox INFO     Loading mailbox subsystem
2020-03-18 00:28:52,341 Repository.Mbox INFO       ↪ loaded invalid mail index: found 3830608 invalid mails
2020-03-18 00:28:52,341 Repository.Mbox INFO     Loading public inboxes
2020-03-18 00:28:55,423 Repository.Mbox INFO       ↪ loaded mail index for alsa-devel@alsa-project.org (shard 0): found 207512 mails
2020-03-18 00:29:01,327 n.pasta_analyse INFO     Regarding mails in time window 2010-01-01--2010-03-01
2020-03-18 00:29:01,327 tory.Repository INFO     Loading mbox commit cache
2020-03-18 00:29:01,329 tory.Repository INFO       ↪ Warning, commit cache file /home/rohit/Projects/PaStA/resources/linux/resources/commit_cache_mbox.pkl not found!
Traceback (most recent call last):
  File "./pasta", line 178, in <module>
    ret = main(deepcopy(sys.argv))
  File "./pasta", line 143, in main
    return analyse(config, sub, argv)
  File "/home/rohit/Projects/PaStA/bin/pasta_analyse.py", line 217, in analyse
    victims = config.psd.commits_on_stacks
  File "/home/rohit/Projects/PaStA/pypasta/Config.py", line 338, in psd
    return self.patch_stack_definition
AttributeError: 'Config' object has no attribute 'patch_stack_definition'
```
Although it should throw an error saying `Analysis mode succ is not available in mailbox mode!` 

This error is due to a bug in the argument parsing sequence in `pasta_analyse.py`

### PR DESCRIPTION
**Pull request by rsarky**
Refactor the if else logic to appropriately fail if succ argument is
passed in mbox mode

Fixes #44

### CODE REVIEW

**Comment by rsarky** [comment]
updated the commit message!

**Comment by rralf** [comment]
Thanks for the PR! But that's a pretty invasive refactoring. I'll have to test and review carefully. Will do that tomorrow. Thank you!

File: bin/pasta_analyse.py [hunk 1]
```diff
@@ -159,69 +159,73 @@ def fill_result(hashes, tag):
         cluster.to_file(f_cluster)
 
     if mbox:
+        if mode == 'succ':
+            log.error('Analysis mode succ is not available in mailbox mode!')
+            return -1
+
```
Comment (rralf) [inline bin/pasta_analyse.py]: This check is already done before, look at line 146.

File: bin/pasta_analyse.py [hunk 2]
```diff
@@ -159,69 +159,73 @@ def fill_result(hashes, tag):
...
+                                characteristic[victim].patches_linux}
+                log.info('Will consider only %u/%u patches (%0.3f%%) as Linux'
+                        'patches' % (len(linux_patches), len(victims),
+                        len(linux_patches) * 100.0 / len(victims)))
+                victims = linux_patches
+                repo.cache_evict_except(victims)
+
+            log.info('Cached %d relevant mails' % len(available))
+            fill_result(victims, False)
+
```
Comment (rralf) [inline bin/pasta_analyse.py:220]: This line contains trailing whitespaces.

**Review by rralf — CHANGES_REQUESTED** [review]
Thanks for the patch. Would please address the comments? Thanks! Ralf

File: bin/pasta_analyse.py (hunk 1 above)
Comment (rsarky) [reply bin/pasta_analyse.py]: forgot to remove my check. fixed now

**Comment by rralf** [comment]
Rebased, and applied to next, with minor stylistic changes. Thanks Rohit!

**Comment by rralf** [comment]
Ok, github is strange. I manually applied the patch to next, and now github thinks I rejected you patch. This is of course not the case.

**Comment by rralf** [comment]
Ah, and nice refactoring. It's really easier to read now.

**Pull request closed without merging**